### PR TITLE
fix: SCRIPT EXISTS must check all nodes

### PR DESCRIPTION
**Pull request by doersf**
`SCRIPT EXISTS <sha>` returns faulty response in cases where a node has restarted and does not have the script in its cache.

### CODE REVIEW

File: lib/redis_client/cluster/router.rb [hunk 1]
```diff
@@ -249,10 +249,22 @@ def send_cluster_command(method, command, args, &block)
...
+        when 'exists'
+          # check for existence on all nodes
+          responses = @node.call_all(method, command, args, &block)
+          final_response = []
+          i = -1
+          loop do
+            i += 1
+            break if responses[0][i].nil?
+
+            final_response << (responses.all? { |r| r[i].eql?(1) } ? 1 : 0)
```
Comment (supercaracal) [inline lib/redis_client/cluster/router.rb:265]: It looks like we can write the logic as a one liner like this:

```
irb(main):013:0> [[1,0],[1,0]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [1, 0]

irb(main):014:0> [[1,0],[0,0]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [0, 0]

irb(main):015:0> [[1,0],[0,1]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [0, 0]

irb(main):016:0> [[0,0],[0,0]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [0, 0]
```

Comment (doersf) [reply lib/redis_client/cluster/router.rb]: ACK. beautiful.

**Comment by supercaracal** [comment]
@doersf Would you mind rebasing your commits?

**Comment by doersf** [comment]
I'll re-open the pull request.

**Pull request closed without merging**